jina/checker.py

```diff
@@ -18,41 +21,64 @@ def __init__(self, args: 'argparse.Namespace'):
         from jina import Client
         from jina.logging.profile import TimeContext
         from jina.serve.runtimes.worker import WorkerRuntime
+        from jina.serve.runtimes.gateway import GatewayRuntime
 
         try:
             total_time = 0
             total_success = 0
-            for j in range(args.retries):
+            for j in range(args.attempts):
                 with TimeContext(
-                    f'ping {args.host} at {j} round', default_logger
+                    f'ping {args.target} on {args.host} at {j} round', default_logger
                 ) as tc:
                     if args.target == 'executor':
-                        r = WorkerRuntime.is_ready(args.host)
+                        hostname, port, protocol, _ = parse_host_scheme(args.host)
+                        r = WorkerRuntime.is_ready(f'{hostname}:{port}')
+                    elif args.target == 'gateway':
+                        hostname, port, protocol, _ = parse_host_scheme(args.host)
+                        r = False
+                        if protocol is None or protocol == 'grpc':
+                            r = GatewayRuntime.is_ready(f'{hostname}:{port}')
+                        else:
+                            try:
+                                conn = urllib.request.urlopen(url=f'http://{hostname}:{port}')
+                                r = (conn.code == HTTPStatus.OK)
+                            except:
+                                r = False
```

Comment:
can we somehow make GatewayRuntime.is_ready apply for all protocols ? this will help us define a unified readiness check for the gateway (to be used in custom gateway PR)